Packages/com.chisel.editor/Chisel/Editor/Editor/Scene/Handles/BoxExtrusionHandle.cs

```diff
@@ -73,36 +84,64 @@ static Bounds GetBounds(bool isSquare, bool generateFromCenter)
 				bounds.Encapsulate(s_Points[1]);
 
 			if (s_Points.Count == 2) return bounds;
-			
-			bounds.Encapsulate(s_Points[0] + (GetHeight(Axis.Y) * Vector3.up));
+
+            var height = GetHeight(axis);
+
+            var size = bounds.size;
+            size[(int)axis] = height;
+            bounds.size = size;
+            
+            if ((flags & BoxExtrusionFlags.GenerateFromCenterY) != BoxExtrusionFlags.GenerateFromCenterY)
+            {
+                var center = bounds.center;
+                center[(int)axis] += height * 0.5f;
+                bounds.center = center;
+            }
 			return bounds;
 		}
+        
+		static bool Inverted(Axis axis)
+		{
+			if (s_Points.Count <= 2) return false;
+			return (s_Points[2] - s_Points[1])[(int)axis] < 0;
+		}
 
 		static float GetHeight(Axis axis)
 		{
 			if (s_Points.Count <= 2) return 0;
 			return (s_Points[2] - s_Points[1])[(int)axis];
 		}
 
-		public static BoxExtrusionState Do(Rect dragArea, out Bounds bounds, out float height, out CSGModel modelBeneathCursor, out Matrix4x4 transformation, bool isSquare, bool generateFromCenter, Axis axis)
+		public static BoxExtrusionState Do(Rect dragArea, out Bounds bounds, out float height, out CSGModel modelBeneathCursor, out Matrix4x4 transformation, BoxExtrusionFlags flags, Axis axis, float? snappingSteps = null)
```

Comment:
the number of options increased enough that using some sort of flags made more sense